dotnet/test/common/BiDi/Network/NetworkEventsTest.cs

```diff
@@ -132,11 +132,8 @@ public async Task CanListenToFetchError()
 
         await using var subscription = await context.Network.OnFetchErrorAsync(tcs.SetResult);
 
-        try
-        {
-            await context.NavigateAsync("https://not_a_valid_url.test", new() { Wait = ReadinessState.Complete });
-        }
-        catch (Exception) { }
+        var navigateTask = context.NavigateAsync("https://not_a_valid_url.test", new() { Wait = ReadinessState.Complete });
+        await ((Task)navigateTask).ConfigureAwait(ConfigureAwaitOptions.SuppressThrowing);
```

Comment:
```suggestion
        await (context.NavigateAsync("https://not_a_valid_url.test", new() { Wait = ReadinessState.Complete }) as Task)
            .ConfigureAwait(ConfigureAwaitOptions.SuppressThrowing);
```

I think writing it as one statement is cleaner, removing extra variable.

Reply:
What do you think of this now? The `as` operator makes me uncomfortable when we know the exact type and don't handle `null`s

Reply:
Actually I don't like this style in tests project, it is harder to understand and maintain. As for me `try/catch` is simpler, we can easily extend of `catch` cases. Probably I am old-school.

But, I see benefit using it in performance critical code execution. There, it would be good.

@YevgeniyShunevych what do you think?

Reply:
I reverted the change here, but I request we keep the changes in `RemoteSeleniumServer`. This test isn't very different with the `SuppressThrowing`. `RemoteSeleniumServer` one is much more simplified (and it used across all tests)

Reply:
I didn't get what is actually going on in the code suggestion, but still. I am absolutely for using `Assert.That(..., Throws...)` construction. But with few suggestions.

For example:

```cs
Assert.That(() =>
{
    _ = driver.CurrentWindowHandle;
}, Throws.InstanceOf<NoSuchWindowException>());
```

I would rewrite this way:

```cs
Assert.That( // No need to wrap delegate body with { }.
    () => driver.CurrentWindowHandle, // No need to add "_ = ". Modern NUnit supports delegates returning values.
    Throws.TypeOf<NoSuchWindowException>()); // Prefer TypeOf instead of InstanceOf to be exact.
```

This is just a style I prefer, you can keep any style you want.

Reply:
I like this style. I went with the first way to minimize diffs, but I am totally in agreement with the second approach.

The code suggestion is using the new `ConfigureAwait(ConfigureAwaitOptions)` method introduced in .NET 8. It allows users to specify how they want the `await` to behave, such as avoiding any exceptions being thrown. Discussions here: https://github.com/dotnet/runtime/pull/87067 https://github.com/dotnet/runtime/issues/22144

I will convert the assertions.

Reply:
> Modern NUnit supports delegates returning values.
I was actually eyeing how much effort it would be to update NUnit to 4.x 😄 

> Prefer TypeOf instead of InstanceOf to be exact.
`InstanceOf` would maintain the existing `catch` behavior, Let me switch to `TypeOf` and make sure nothing breaks

Reply:
Feedback addressed!